### DISCUSSION TITLE
IOS download external SWFs

### DISCUSSION THREAD

**Discussion by talbenarie** [airsdk/Adobe-Runtime-Support#2445] [Q&A]
My application relies on downloading external swf files and manually caching them in the app. However, in iOS devices it seem like i am unable to do so. It works in mac, android, pc and web but in ios version it seems like it is not working. Perhaps there is a missing permission that needs to be granted, or something is wrong when pubishing the app? 
Note: im using test flight to test the app.
Would appreciate any help

**Comment by shaucky** [comment]
According to many referance about AS3, AIR does not support loading SWF files containing ABC on iOS after the application is packaged.

If the target platform is iOS, the compiler compiles ActionScript directly into native code for iOS.

Some related referance goes here: https://airsdk.dev/reference/actionscript/3.0/flash/display/Loader.html

**Reply by talbenarie** [reply]
I have read that there are ways to do so with SWFPreloader or starling but no idea how

**Reply by ajwfrost** [reply]
On iOS, the only way you can load external SWF files is where they've been pre-processed by the ADT tool when you build the initial IPA file. This is because of Apple's restriction in not allowing any script that is interpreted/JIT-compiled, other than their own JavaScriptCore stuff. But if you include the SWFs in the `externalSwfs` file list, you can then load the pre-processed versions of these from your remote server at run-time..

A few other references; sadly it looks like the original guidelines from the Adobe blog have gone now..
https://airsdk.dev/docs/building/application-descriptor-files/elements/iPhone#externalswfs
https://stackoverflow.com/questions/15411603/air-3-6-ios-load-external-swf-with-abc
https://forum.starling-framework.org/d/16451-ios-loading-assets-from-an-external-stripped-swf/22

thanks

**Reply by ajwfrost** [reply]
Yes, they have to be stripped of any ActionScript ByteCode blocks, otherwise the runtime throws an error.
